src/fe/inf_fe_map.C

```diff
@@ -27,11 +27,43 @@
 #include "libmesh/inf_fe_macro.h"
 #include "libmesh/libmesh_logging.h"
 
+#include "libmesh/dense_matrix.h"
+#include "libmesh/analytic_function.h" // for DenseVector
+
 namespace libMesh
 {
 
+   // local helper-function solving Ax=b in 3D.
+bool system_solve_3x3(DenseMatrix<libMesh::Real> & A, DenseVector<libMesh::Real> & b, DenseVector<libMesh::Real> & x)
```

Comment:
If it's possible to use a `TypeTensor` here instead of a `DenseMatrix`, there is already a `solve()` member on that. It would be good not to repeat this code for solving a 3x3 system if possible.

Reply:
great, I was not aware of this.